sklearn/metrics/cluster/unsupervised.py

```diff
@@ -205,6 +209,8 @@ def silhouette_samples(X, labels, metric='euclidean', **kwds):
 
     sil_samples = inter_clust_dists - intra_clust_dists
     sil_samples /= np.maximum(intra_clust_dists, inter_clust_dists)
```

Comment:
Also if they belong to different cluster. Quick example.

```python
 silhouette_samples([[1.0], [1.0], [1.0], [1.0]], [0, 0, 1, 1])
```

However, according to the formula, this might be expected behaviour.